jax/_src/lax/parallel.py

```diff
@@ -730,6 +740,82 @@ def pos_reduce(x):
 core.axis_substitution_rules[pmin_p] = partial(_subst_all_names_in_param, 'axes')
 
 
+# TODO(mattjj): merge with lax.argmin_p / lax.argmax_p
+pargmax_p = core.Primitive('pargmax')
+pargmin_p = core.Primitive('pargmin')
+
+def _pargminmax_impl(prim, x, *, axis):
+  if not isinstance(axis, int): raise NameError(f"unbound axis name: {axis}")
+  return xla.apply_primitive(prim, x, axis=axis)
+pargmax_p.def_impl(partial(_pargminmax_impl, pargmax_p))
+pargmin_p.def_impl(partial(_pargminmax_impl, pargmin_p))
+
+def _pargminmax_abstract_eval(x, *, axis):
+  if isinstance(axis, int):
+    if not 0 <= axis < np.ndim(x):
+      raise ValueError(f"axis index {axis} for operand shape {np.shape(x)}")
+    shape = tuple(np.delete(np.shape(x), axis))
+  else:
+    # TODO(mattjj): avals with names, check and eliminate axis name
+    shape = np.shape(x)
+  return ShapedArray(shape, np.dtype('int32'))
+pargmax_p.def_abstract_eval(_pargminmax_abstract_eval)
+pargmin_p.def_abstract_eval(_pargminmax_abstract_eval)
+
+def _pargminmax_vmap_collective_rule(reducer, frame, vals_in, dims_in, *, axis):
+  del frame, axis  # Unused.
+  (x,), (d,) = vals_in, dims_in
+  return reducer(x, d, np.dtype('int32')), None
+batching.collective_rules[pargmax_p] = partial(_pargminmax_vmap_collective_rule,
+                                               lax.argmax)
+batching.collective_rules[pargmin_p] = partial(_pargminmax_vmap_collective_rule,
+                                               lax.argmin)
+
+def _pargminmax_vmap_batching_rule(prim, vals_in, dims_in, *, axis):
+  (x,), (d,) = vals_in, dims_in
+  if isinstance(axis, int):
+    new_axis = axis + (d <= axis)
+    return prim.bind(x, axis=new_axis)
+  else:
+    return prim.bind(x, axis)
```

Comment:
Doesn't this also have to return a second component for the batch axis in the output? How come aren't the tests catching this?

Reply:
Good catch, I forgot to add a test for this case! I should add an xmap test that uses systematic schedules.